### DISCUSSION TITLE
How do i write asp.net core websites inside of vs 2019?

### DISCUSSION THREAD

**Discussion by zedjr** [OrchardCMS/OrchardCore#10168] [General]
I have a site written in asp.net core 3.1 or 5 and i want to learn orchard core. I am thinking this will be a decoupled site and want to transfer it to a orchard core cms so something can be easily edited from admin panel. This is like a beginner project so i can learn orchard they way i need to use it. I see tutorials but they are inside of using the admin panel. SO many more advantages writing code inside of VS 2019. Intellisense for one and I don't want to work this way. I see the one tutorial from orchard site for a decoupled site but am totally lost by minute 1. Not showing or explaining on how to set it up or anything. I though honestly I could just add my own service , configure it and then just use tag wrappers around my html or something that needs to be edited by user. That is really my first set of goals in learning orchard. If anyone can direct me in the write direction I would appreciate it.

**Comment by hishamco** [comment]
FYI Orchard Core support multi-targeting 3.1 & 5.0, it runs fine in VS 2019

**Comment by hishamco** [comment]
You could start with default ASP.NET Core project template, then add an OC module. There's a rich docs in http://docs.orchardcore.net/ also you can find some videos on YouTube by @sebastienros explained OC in ASP.NET Community Standup

**Comment by ns8482e** [comment]
To get started with decoupled site follow this step by step guide - https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/

**Reply by zedjr** [reply]
ok i see the example is using razor pages. Do i fllow this guide first then the guide on "Create a Modular Application". I want to write my own controller and views and where i need manageable content add a module i guess? It is jsut so confusing to me the guide on create a modular application and then this guide. Which is what?

**Reply by ns8482e** [reply]
The guide uses Razor Pages as an example, However nothing stops you to use Controller/Views  instead of Razor Pages.

`decoupled-cms` guide is for using `Orchard Core CMS` in decoupled way -   https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/
Follow this guide if you are building modular application  that needs content management, user management and admin functionality.

`Create a Modular Application` https://docs.orchardcore.net/en/latest/docs/guides/create-modular-application-mvc/  this guide is for using `Orchard Core Framework`  in `MVC`.  Follow this guide if you need to add modularity  in your MVC application.

**Reply by zedjr** [reply]
> The guide uses Razor Pages as an example, However nothing stops you to use Controller/Views instead of Razor Pages.
> 
> `decoupled-cms` guide is for using `Orchard Core CMS` in decoupled way - https://docs.orchardcore.net/en/latest/docs/guides/decoupled-cms/
> Follow this guide if you are building modular application that needs content management, user management and admin functionality.
> 
> `Create a Modular Application` https://docs.orchardcore.net/en/latest/docs/guides/create-modular-application-mvc/ this guide is for using `Orchard Core Framework` in `MVC`. Follow this guide if you need to add modularity in your MVC application.

ok thank you. As easy as that sounds that helps in my state of confusion. So am curious why is there a section separately just for mvc it seems labeled " Create a Modular Application".
